src/controllers/connectionManager.ts

```diff
@@ -447,74 +442,56 @@ export default class ConnectionManager {
-            }
-
-            // Get list of databases on current server
-            let listParams = new ConnectionContracts.ListDatabasesParams();
-            listParams.ownerUri = fileUri;
-            self.client.sendRequest(ConnectionContracts.ListDatabasesRequest.type, listParams).then((result: any) => {
-                // Then let the user select a new database to connect to
-                self.connectionUI.showDatabasesOnCurrentServer(self._connections[fileUri].credentials, result.databaseNames).then( newDatabaseCredentials => {
-                    if (newDatabaseCredentials) {
-                        self.vscodeWrapper.logToOutputChannel(
-                            Utils.formatString(LocalizedConstants.msgChangingDatabase, newDatabaseCredentials.database, newDatabaseCredentials.server, fileUri)
-                        );
-
-                        self.disconnect(fileUri).then( () => {
-                            self.connect(fileUri, newDatabaseCredentials).then( () => {
-                                self.vscodeWrapper.logToOutputChannel(
-                                    Utils.formatString(
-                                        LocalizedConstants.msgChangedDatabase,
-                                        newDatabaseCredentials.database,
-                                        newDatabaseCredentials.server, fileUri)
-                                );
-                                resolve(true);
-                            }).catch(err => {
-                                reject(err);
-                            });
-                        }).catch(err => {
-                            reject(err);
-                        });
-                    } else {
-                        resolve(false);
-                    }
-                }).catch(err => {
-                    reject(err);
-                });
-            });
-        });
-    }
-
-    public async changeDatabase(newDatabaseCredentials: IConnectionCredentials): Promise<boolean> {
-        const self = this;
-        const fileUri = this.vscodeWrapper.activeTextEditorUri;
-        return new Promise<boolean>(async (resolve, reject) => {
-            if (!self.isConnected(fileUri)) {
-                self.vscodeWrapper.showWarningMessage(LocalizedConstants.msgChooseDatabaseNotConnected);
-                resolve(false);
-                return;
-            }
+        // Get list of databases on current server
+        let listParams = new ConnectionContracts.ListDatabasesParams();
+        listParams.ownerUri = fileUri;
+        const result = await self.client.sendRequest(ConnectionContracts.ListDatabasesRequest.type, listParams);
+        // Then let the user select a new database to connect to
+        const newDatabaseCredentials = await self.connectionUI.showDatabasesOnCurrentServer(self._connections[fileUri].credentials, result.databaseNames);
+        if (newDatabaseCredentials) {
+            self.vscodeWrapper.logToOutputChannel(
```

Comment:
Self is probably just making this code more complicated to understand now. 

But not really an issue.